sphinx/builders/latex/__init__.py

```diff
@@ -510,25 +510,25 @@ def setup(app: Sphinx) -> Dict[str, Any]:
     app.connect('config-inited', validate_latex_theme_options, priority=800)
     app.connect('builder-inited', install_packages_for_ja)
 
-    app.add_config_value('latex_engine', default_latex_engine, None,
+    app.add_config_value('latex_engine', default_latex_engine, False,
```

Comment:
i haven't changed the type signature of this method, i've only changed the way it's called to *comply* with the pre-existing type signature.
I can absolutely do what you're describing, by changing the signature from

```python
    def add_config_value(self, name: str, default: Any, rebuild: Union[bool, str],
                         types: Any = ()) -> None:
        ...
```

to 

```python
    def add_config_value(self, name: str, default: Any, rebuild: str,
                         types: Any = ()) -> None:
        ...
```

but this is a breaking change, and i think out of the scope of this PR.
Looking at the code, a literal boolean is perfectly acceptable, so it could be that the docstring is out of date.

there's also this line-

```
        .. versionchanged:: 0.6
           Changed *rebuild* from a simple boolean (equivalent to ``''`` or
           ``'env'``) to a string.  However, booleans are still accepted and
           converted internally.
```

so the docstring is consistent with the usage.

Is there a plan to deprecate the boolean arg?

Reply:
hi @tk0miya how would you like to proceed here? I can change all the uses of `bool` to `str` for this method if you prefer? Are there plans then to deprecate `bool`? If so, what's the most appropriate way to mark this?

something like this-

```python
import warnings
warnings.warn("blah blah blah", DeprecationWarning)
```
?

Reply:
@tk0miya i'm going to leave it for now and not address this in this PR. The reason being is i think it's a bigger change, and requires separate discussion.

Already this method doesn't actually accept any old string, it only accepts string literals of either `env` or `html`. Strictly speaking this should probably use an `Enum`, but the next best choice (retaining backwards compatibility) is to use `typing.Literal`.